admin/vlaudit-stats/vlaudit-stats

```diff
@@ -0,0 +1,668 @@
+                self.msgid_weird(self.last_msgid, msgid, self.last_tstamp, tstamp)
+            elif diff > 1:
+                self.msgid_gap(diff - 1, self.last_tstamp, tstamp)
+
+        self.last_msgid = msgid
+        self.last_tstamp = tstamp
+
+    def bad_line(self, line):
+        self.n_bad += 1
+        self.last_bad = self.now()
+
+    def getent(self, req):
+        self.ips.hit(req.host, req)
+        if req.code == 0:
+            self.pos.hit(req.reqvol, req)
+        else:
+            self.neg.hit(req.reqvol, req)
+
+    def restarted(self, tstamp):
+        self.n_restarts += 1
+        self.last_restart = self.now()
+        self.last_restart_ts = tstamp
+
+    def msgid_weird(self, from_id, to_id, from_ts, to_ts):
+        self.audit_n_weird += 1
+        self.audit_weird_from = from_id
+        self.audit_weird_to = to_id
+        self.audit_weird_last = self.now()
+        self.audit_weird_fromts = from_ts
+        self.audit_weird_tots = to_ts
+        warn("msgid went backwards: %d -> %d (audit tstamp %s -> %s)" % (
+             from_id, to_id, from_ts, to_ts))
+
+    def msgid_gap(self, n_miss, from_ts, to_ts):
+        self.audit_n_miss += n_miss
+        self.audit_gaps.append(AuditGap(n_miss, from_ts, to_ts, self.now()))
+
+class AuditGap:
+    def __init__(self, n_miss, from_ts, to_ts, now):
+        self.n_miss = n_miss
+        self.from_ts = from_ts
+        self.to_ts = to_ts
+        self.secs = now
+
+    def stats(self):
+        return dict(
+            n_miss=self.n_miss,
+            from_ts=self.from_ts,
+            to_ts=self.to_ts,
+            now=self.secs
+        )
+
+# The data for a single VL GetEntry request (as reported by the audit log).
+class GetEntryRequest:
+    def __init__(self, tstamp, match):
+        self.tstamp = tstamp
+
+        (code,
+         self.host,
+         self.reqvol,
+         volid,
+         self.volname) = match.group(1,2,4,5,6)
+
+        self.code = int(code)
+        self.volid = int(volid)
+
+    def stats(self):
+        ret = dict(
+            tstamp=self.tstamp,
+            host=self.host,
+            code=self.code,
+            vol_requested=self.reqvol,
+        )
+        if self.code == 0:
+            ret['vol_id'] = self.volid
+            ret['vol_name'] = self.volname
+        return ret
+
+# Our reader and parser of the VL audit log.
+class AuditReader:
+    # Match groups:
+    # 1: code
+    # 2: host
+    # 3: "STR" (for GetByName) or "LONG" (for GetByID)
+    # 4: requested vol name/id
+    # 5: rwid for returned volume
+    # 6: name for returned volume
+    #
+    # We don't really distinguish between GetEntryByID and GetEntryByName
+    # requests in here. If we wanted to in the future, just check for
+    # reqvol.isdigit() to see if it's a numeric request.
+    getent_pat = re.compile(r'^\[\d+\] ... ... \d\d \d\d:\d\d:\d\d \d{4} EVENT AFS_VL_GetEnt CODE ([0-9]+) NAME .* HOST ([^ ]+) (STR|LONG) (.*) LONG ([0-9]+) STR (.*) $')
```

Comment:
openafs gerrit 14467 has changed the relevant audit names here. I have this fixed locally, but I won't push the change here yet, in case 14467 changes any more, or if this PR yields any additional comments in the meantime :)